SW-2130 / SW-2304 Add photos to withdrawal

karthikbtf opened the pull request:
- add component to select photos
- refactor withdrawal flow to host tittle and buttons in parent page
- use validate flag to control flow progression
- refactor nursery withdrawal data model population
- create withdrawal with photo uploads
- TODOs are marked as TODOs, Jiras have been created for those

karthikbtf commented:
Confirmed it provides an option to load a photo vs browse files on mobile.
This one is good to go.

karthikbtf commented on src/components/OptInFeatures/index.tsx:
```diff
@@ -93,6 +95,14 @@ export default function OptInFeatures({ refresh }: OptInFeaturesProps): JSX.Elem
     color: isDisclosure ? theme.palette.TwClrTxtDanger : 'inherit',
   });
 
+  /**
+   * REMOVE
+   */
+  const onPhotosChanged = (files: File[]) => {
+    // tslint:disable-next-line: no-console
+    console.log('onPhotosChanged', files);
+  };
+
```
this will be removed, just for testing in vercel

karthikbtf commented on src/components/OptInFeatures/index.tsx:
```diff
@@ -157,6 +167,13 @@ export default function OptInFeatures({ refresh }: OptInFeaturesProps): JSX.Elem
               </Grid>
             </Stack>
           ))}
+          <Box display='flex' margin='50px auto' maxWidth={800}>
+            <SelectPhotos
+              title={strings.ADD_PHOTOS_REQUIRED}
+              description={strings.ADD_PHOTOS_DESCRIPTION}
+              onPhotosChanged={onPhotosChanged}
+            />
+          </Box>
```
will be removed prior to checkin

karthikbtf commented on src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx:
```diff
@@ -152,17 +120,23 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
-    setRecords(localRecords.filter((rcd) => rcd.facilityId.toString() === selectedNursery));
-
-    onNext();
+    onNext({
```
@constanzauanini looks like we got something wrong earlier.
The withdrawal is only one object, not an array. 
The batchWithdrawals property within the withdrawal object is the filtered down array of batches.
Fixed the code, it is actually a lot simpler now.

karthikbtf commented on src/components/Inventory/withdraw/BatchWithdrawFlow.tsx:
```diff
@@ -63,7 +71,8 @@ export default function BatchWithdrawFlow(props: BatchWithdrawFlowProps): JSX.El
   }, [batchIds]);
 
-  const onPurposeSelected = () => {
+  const onWithdrawalConfigured = (withdrawal: NurseryWithdrawal) => {
```
updated this to take in the new value of withdrawal with filtered down batches.
ideally we don't want to be passing setStates across components, each component should ideally support an onChangexyz and allow the parent to control its own state.

karthikbtf commented on src/components/Inventory/withdraw/BatchWithdrawFlow.tsx:
```diff
@@ -19,7 +21,13 @@ type BatchWithdrawFlowProps = {
   const { organization, batchIds } = props;
   const [flowState, setFlowState] = useState<FlowStates>('purpose');
-  const [records, setRecords] = useForm<NurseryWithdrawal[]>([]);
+  const [record, setRecord] = useForm<NurseryWithdrawal>({
```
this is the source withdrawal object, it's a single withdrawal with a bulk of batches. we got it wrong earlier, i messed up a bit. the array is only on the batchWithdrawals property which contains batch specific info.

constanzauanini replied:
Excellent!

karthikbtf reviewed (commented):
Refactored the flow management a bit to have a parent flow controlling the cancel/next buttons and instantiating child component within the header/footer of parent component.

charlie-tf reviewed (approved):
Works great! Tested Vercel on desktop and mobile. Code is clean and easy to follow

karthikbtf commented:
> Works great! Tested Vercel on desktop and mobile. Code is clean and easy to follow

Thanks @charlie-tf ! I reverted one refactor to simplify things - and renamed a file since your last review. If you can take a look again lmk.

karthikbtf commented:
@constanzauanini I got withdrawals working with photo uploads.
Had to refactor some code (see PR description).
Outplant is yet to be supported, will work on that next. Same for formatting the success message.
The bulk withdraw quantity setting is to be done - as per plan will leave that for you.
Will merge this in and happy to follow up if you have comments.
